modules/social_features/social_profile/modules/social_profile_privacy/social_profile_privacy.module

```diff
@@ -229,6 +257,21 @@ function _social_profile_privacy_fields_submit($form, FormStateInterface $form_s
   }
 
   \Drupal::service('cache_tags.invalidator')->invalidateTags($tags);
+
+  if (!isset($profile)) {
+    return;
+  }
+  /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+  $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+  if (!empty($profile_storage)) {
```

Comment:
Is it a good idea to check if the name fields are changed or not and then continue with updating the profiles?

Reply:
In this place, I can check only the visibility settings of fields for the current account (see screenshot). Is it enough or we need to check updating Profile name fields values?

<img width="772" alt="Screenshot 2021-05-31 at 11 35 39" src="https://user-images.githubusercontent.com/10220937/120165518-56643e80-c204-11eb-95af-7b5e44dc3b11.png">

Reply:
@Ressinel Yes, that would be a good idea.

Reply:
I have implemented it.